packages/server/src/api/controllers/deploy/index.ts

```diff
@@ -98,13 +101,25 @@ async function initDeployedApp(prodAppId: any) {
   console.log("Enabled cron triggers for deployed app..")
 }
 
-async function deployApp(deployment: any) {
+async function deployApp(deployment: any, userId: string) {
   let replication
   try {
     const appId = getAppId()
     const devAppId = getDevelopmentAppID(appId)
     const productionAppId = getProdAppID(appId)
 
+    // can't do this in test
+    if (!env.isTest()) {
```

Comment:
I think we should start straying away from the `isTest` function, as it causes our test and production paths to diverge.
We might be better mocking the `triggerAppBackup` function instead, so our test can assert this is called with the right params.

Reply:
Sadly this would be a nightmare - I think its better to keep using the `isTest` for the time being, its a great marker for stuff that we need to update as well! Mocking out the entirety of the object store would be painful as it involves file system access, and would all just be ripped out next month anyway.

Reply:
I don't think we need to mock object store here, just mock pro by doing 

```ts
const pro = jest.requireActual("@budibase/pro")
jest.mock("@budibase/pro", () => ({
  ...pro,
  backups: {
    triggerAppBackup: jest.fn(),
  },
}))
```
jest.mock(

Reply:
Added in: https://github.com/Budibase/budibase/pull/8354/commits/0274b6755a20ec330c8a6321108f851e615ac43c